pysidt/sidt.py

```diff
@@ -317,19 +357,42 @@ def write_nodes(tree, file):
             p = None
         else:
             p = node.parent.name
+            
+        try:
+            json.dumps(node.rule)
+            rule = node.rule
+        except (TypeError, OverflowError):
+            rule = to_dict(node.rule) #will work on all rmgmolecule objects, new objects need this method implemented
+            try:
+                json.dumps(rule)
+            except:
+                print(rule)
```

Comment:
Can you write proper logger and informative error message

Reply:
I've provided a more informative error message and removed the logging.